src/http/server/handlers/static_file_handler.cr

```diff
@@ -5,9 +5,9 @@ class HTTP::StaticFileHandler < HTTP::Handler
   end
 
   def call(request)
-    file_path = @publicdir + request.path
+    file_path = @publicdir + request.resource
     if Dir.exists?(file_path)
-      HTTP::Response.new(200, directory_listing(request.path, file_path), HTTP::Headers{"Content-Type": "text/html"})
+      HTTP::Response.new(200, directory_listing(request.resource, file_path), HTTP::Headers{"Content-Type": "text/html"})
```

Comment:
You are right it should just be #path here
On Sep 23, 2015 9:39 AM, "Julien Portalier" <notifications@github.com>
wrote:

> In src/http/server/handlers/static_file_handler.cr
> <https://github.com/manastech/crystal/pull/1564#discussion_r40177166>:
>
> >      if Dir.exists?(file_path)
> > -      HTTP::Response.new(200, directory_listing(request.path, file_path), HTTP::Headers{"Content-Type": "text/html"})
> > +      HTTP::Response.new(200, directory_listing(request.resource, file_path), HTTP::Headers{"Content-Type": "text/html"})
>
> Minor detail: shouldn't it target the file at the request path, skipping
> the query params?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/manastech/crystal/pull/1564/files#r40177166>.
>